dart/lib/flat_buffers.dart

```diff
@@ -330,20 +330,22 @@ class Builder {
     return tableTail;
   }
 
-  /// This method low level method can be used to return a raw piece of the buffer
-  /// after using the the put* methods.
+  /// Returns the current buffer. In most cases, you should only use this after
+  /// calling [finish()] with the root table offset.
```

Comment:
What are the cases where it's ok to not do that?

IOW, why aren't we documenting this as `You must call [finish] before accessing this getter`, and asserting that finish has been called?

Reply:
the previous function (`lowFinish()` provided the access to the buffer and is used from tests. Didn't want to take away that option

Reply:
It seems awkward to me to expose the buffer of an unfinished builder - maybe we can just rewrite the tests to assert the buffer produced by finish is correct?

IOW this seems like an API surface that is only used to test internals that probably don't need to be tested, and there's no good static way to analyze whether it's being used correctly or not.

Reply:
I'm fine with that, you want me to change that?

Reply:
Yes please

Reply:
Changed. After #6711 is merged, I'll be able to fix the `test_low()` while keeping the logic of testing the low-level/internal buffer.

Reply:
rebased and updated the test - ready to merge from my POV